ethereum/core/src/main/java/org/hyperledger/besu/ethereum/transaction/TransactionSimulator.java

```diff
@@ -291,6 +283,38 @@ public Optional<TransactionSimulatorResult> processWithWorldUpdater(
     return Optional.of(new TransactionSimulatorResult(transaction, result));
   }
 
+  private long calculateSimulationGasCap(
+      final CallParameter callParams, final BlockHeader blockHeaderToProcess) {
```

Comment:
Sorry to come back to you on this, but rather than passing whole `CallParameter` and `BlockHeader`, we can pass gas limits only, it would make it even more focused